gcapi/client.py

```diff
@@ -640,24 +644,34 @@ def upload_cases(  # noqa: C901
 
         return raw_image_upload_session
 
-    def run_external_job(  # noqa: C901
-        self, *, algorithm: str, inputs: dict[str, Any]
+    def run_external_job(
+        self,
+        *,
+        algorithm: Union[str, gcapi.models.Algorithm],
+        inputs: SocketValueSetDescription,
     ):
         """
         Starts an algorithm job with the provided inputs.
+
         You will need to provide the slug of the algorithm. You can find this in the
         url of the algorithm that you want to use. For instance, if you want to use
         the algorithm at
             https://grand-challenge.org/algorithms/corads-ai/
         the slug for this algorithm is "corads-ai".
-        For each input interface defined on the algorithm you need to provide a
-        key-value pair (unless the interface has a default value),
-        the key being the slug of the interface, the value being the
-        value for the interface. You can get the interfaces of an algorithm by calling
+
+        For each input socket defined on the algorithm you need to provide a
+        key-value pair (unless the socket has a default value),
+        the key being the slug of the socket, the value being the
+        value for the socket. You can get the interfaces
+        (i.e. all possible socket sets) of an algorithm by
+        calling
             client.algorithms.detail(slug="corads-ai")
-        and inspecting the ["inputs"] of the result.
-        For image type interfaces (super_kind="Image"), you can provide a list of
+
+        and inspecting the ["interfaces"] of the result.
+
+        For image-kind sockets (super_kind="Image"), you can provide a list of
         files, which will be uploaded, or a link to an existing image.
+
```

Comment:
Maybe give an example that reuses an existing CIV and how the user would pass that in?

Reply:
Can we actually use existing CIVs? How would that work? Adding a api_url to the CIV?

Reply:
The new thing that that allows is linking existing non-image files (even though they will still be re-uploaded), but so far the user had to have them locally first, now gcapi does the download for the user. Right?

Reply:
> I was referring to what you explained in the create strategies PR [here](https://github.com/DIAGNijmegen/rse-gcapi/pull/173#discussion_r2050612685). Did I misunderstand that?

Nope. Indeed. That works. I was thinking how to do the 'run from archive' efficiently via the gcapi. I thought perhaps we could point to existing CIVs directly: but that is not the case atm.

Reply:
I've heavily updated the docstrings, correcting some naming and indentation. The Visual Studio Code within IDE now renders them nicely.

Reply:
What do you mean that is not the case at the moment? A user could loop over all archive items in an archive and then schedule a job for each by supplying the values from the archive item as inputs to the job:
`client.run_external_job(
                algorithm="corads-ai",
                inputs=archive_item["values"]
                )`